l2.py

```diff
@@ -0,0 +1,234 @@
+        try:
+            lib.run(
+                "generating L2 genesis and rollup configs",
+                ["go", "run", "cmd/main.go", "genesis", "l2",
+                    "--l1-rpc=http://localhost:8545",
+                    f"--deploy-config={paths.network_config_path}",
+                    f"--deployment-dir={paths.devnet_l1_deployment_dir}",
+                    f"--outfile.l2={paths.l2_genesis_path}",
+                    f"--outfile.rollup={paths.rollup_config_path}"],
+                cwd=paths.op_node_dir)
+        except Exception as err:
+            raise lib.extend_exception(err, prefix="Failed to generate L2 genesis and rollup configs: ")
+
+
+####################################################################################################
+
+
+class DevnetL2Config:
+    def __init__(self, geth_data_dir: str, paths: OPPaths):
+        self.data_dir = geth_data_dir
+        """Geth data directory for devnet op-geth node."""
+
+        self.keystore_dir = f"{self.data_dir}/keystore"
+        """Keystore directory for devnet op-geth node (each file stores an encrypted signer key)."""
+
+        self.chaindata_dir = f"{self.data_dir}/geth/chaindata"
+        """Directory storing chain data."""
+
+        self.password_path = f"{self.data_dir}/password"
+        """Path to file storing the password for the signer key."""
+
+        self.password = "l2_devnet_password"
+        """Password to use to secure the signer key."""
+
+        self.tmp_signer_key_path = f"{self.data_dir}/block-signer-key"
+        """Path to file storing the signer key during the initial import."""
+
+        self.signer_address = "0xca062b0fd91172d89bcd4bb084ac4e21972cc467"
+        """Address of the block signer."""
+
+        self.signer_private_key = "3e4bde571b86929bf08e2aaad9a6a1882664cd5e65b96fff7d03e1c4e6dfa15c"
+        """Private key of the block signer."""
+
+        genesis = lib.read_json_file(paths.l2_genesis_path)
+        self.chain_id = genesis["config"]["chainId"]
+
+        self.jwt_secret_path = paths.ops_bedrock_dir + "/test-jwt-secret.txt"
+        """Path for Jason Web Token secret, used for devnet op-geth rpc auth."""
+
+        # For the following values, allow environment override for now, to follow the original.
+        # In due time, remove that as we provide our own way to customize.
+
+        self.verbosity = os.environ.get("GETH_VERBOSITY", 3)
+        """Geth verbosity level (from 0 to 5, see geth --help)."""
+
+        self.rpc_port = os.environ.get("RPC_PORT", 9545)
+        """Port to use for the http-based JSON-RPC server."""
+
+        self.ws_port = os.environ.get("WS_PORT", 9546)
+        """Port to use for the WebSocket-based JSON_RPC server."""
+
+
+####################################################################################################
+
+def start_devnet_l2_node(paths: OPPaths):
+    """
+    Spin the devnet op-geth node, then wait for it to be ready.
+    """
+
```

Comment:
The version of the OP repo we were using used the same entrypoint shell script for L1 and L2, but quite a few of the options are not actually needed on L2. The lastest version of the Optimism repo has them split out:

- https://github.com/ethereum-optimism/optimism/blob/9c1ee08c0606d067a9754c4b6d798933fbe01d3a/ops-bedrock/entrypoint-l1.sh
- https://github.com/ethereum-optimism/optimism/blob/9c1ee08c0606d067a9754c4b6d798933fbe01d3a/ops-bedrock/entrypoint-l2.sh

Will comment as appropriate!

Reply:
I also see that upnode has added the following options:

```
  --rollup.disabletxpoolgossip=true \
  --txlookuplimit=0
```

(from https://github.com/upnodedev/op-stack-basic-deployment/blob/main/entrypoints/op-geth.sh)

We should investigate, see what these do, if they exist in the version we use.

Reply:
txlookuplimit default value is one year which i think it is enough